src/Common/ColumnsHashing.h

```diff
@@ -378,6 +378,22 @@ struct HashMethodSingleLowCardinalityColumn : public SingleColumnMethod
             return EmplaceResult(inserted);
     }
 
+    template <typename Data>
+    ALWAYS_INLINE FindResult findKey(Data & data, size_t row_, Arena & pool)
```

Comment:
This implementation does not use `saved_hash`, and `visit_cache`, actually below there is right method `findFromRow`, but probably functions signatures were changed and now it is not used.
In FindResult offsets where added here https://github.com/ClickHouse/ClickHouse/pull/20079/files, and `findFromRow` seems a bit outdated, we need to rename method to `findKey` and then fix compile time issues.
I am not sure that I get it right:
```c++
template <typename Data>
    ALWAYS_INLINE FindResult findKey(Data & data, size_t row_, Arena & pool)
    {
        size_t row = getIndexAt(row_);

        if (is_nullable && row == 0)
        {
            if constexpr (has_mapped)
                return FindResult(data.hasNullKeyData() ? &data.getNullKeyData() : nullptr, data.hasNullKeyData(), 0);
            else
                return FindResult(data.hasNullKeyData(), 0);
        }

        if (visit_cache[row] != VisitValue::Empty)
        {
            if constexpr (has_mapped)
                return FindResult(&mapped_cache[row], visit_cache[row] == VisitValue::Found, 0);
            else
                return FindResult(visit_cache[row] == VisitValue::Found, 0);
        }

        auto key_holder = getKeyHolder(row_, pool);

        typename Data::LookupResult it;

        if (saved_hash)
            it = data.find(keyHolderGetKey(key_holder), saved_hash[row]);
        else
            it = data.find(keyHolderGetKey(key_holder));

        bool found = it;
        visit_cache[row] = found ? VisitValue::Found : VisitValue::NotFound;

        if constexpr (has_mapped)
        {
            if (found)
                mapped_cache[row] = it->getMapped();
        }

        size_t offset = 0;

        if constexpr (FindResult::has_offset)
            offset = found ? data.offsetInternal(it) : 0;

        if constexpr (has_mapped)
            return FindResult(&mapped_cache[row], found, offset);
        else
            return FindResult(found, offset);
    }
```